### DISCUSSION TITLE
Gracefull sleep and wakeup mode

### DISCUSSION THREAD

**Discussion by devprofile98** [MaJerle/lwcell#88] [Q&A]
Hello, Thank you for this great library. Is there any specific sequence of function calls to make the SIM device **asleep** ( or **minimum functionality mode** ) and ensure all of the library parts are aware of this? 

in my first try, I called `lwgsm_set_func_mode(0, ...)` but after `calling lwgsm_set_func_mode(1, ...)`, there will be a **failed assertion** on one of the **MQTT semaphore**, I wanted to ensure if there is a known path for the sleep mode to work correctly and if it is implemented already or if it depends on how the client has used the LWCELL library.

Thank you very much.

**Comment by MaJerle** [comment]
Disabling the functionality is not a modem sleep, it is rather a modem "shutdown" mode. In this mode, everything is closed, even your MQTT connections.

**Reply by devprofile98** [reply]
Thank you. but, Can `lwgsm_set_func_mode(0, ...)` be used in the middle of a program, or is its use restricted to certain points in the code? Are there specific considerations or related function calls that must be respected when using it?

**Reply by MaJerle** [reply]
You can call it in the middle, but I don't see a true point of doing it. What's the task you'd like to achieve?

**Reply by devprofile98** [reply]
I am testing various options for reducing the power consumption of the SIM modem. We encountered technical issues with the `AT+CSCLK` command for entering actual sleep mode, so we are exploring the use of minimum functionality mode. According to the [documentation](https://docs.rs-online.com/2190/0900766b815ca94b.pdf), the current consumption for both modes is similar.
